src/Microsoft.Extensions.Logging.Abstractions/Internal/LogValuesFormatter.cs

```diff
@@ -17,25 +17,47 @@ public class LogValuesFormatter
     {
         private const string NullValue = "(null)";
         private static readonly object[] EmptyArray = new object[0];
-        private static readonly char[] FormatDelimiters = {',', ':'};
-        private readonly string _format;
-        private readonly List<string> _valueNames = new List<string>();
+        private static readonly char[] FormatDelimiters = { ',', ':' };
+        private string _format;
+        private List<string> _valueNames;
 
         public LogValuesFormatter(string format)
         {
             OriginalFormat = format;
+        }
 
-            var sb = new StringBuilder();
+        public string OriginalFormat { get; private set; }
+        public List<string> ValueNames
+        {
+            get
+            {
+                if (string.IsNullOrEmpty(_format)) FormatInput(OriginalFormat); //construct the valuenames from the input
+                return _valueNames ?? new List<string>(0);
+            }
+        }
+
+        private string FormatInput(string format)
+        {
+            if (!string.IsNullOrEmpty(_format))
+            {
+                return _format;
+            }
             var scanIndex = 0;
             var endIndex = format.Length;
+            char openBrace = '{', closingBrace = '}';
 
+            //{} => Nothing to process, min length should be greater than 2. Ex: "{0}"
+            if ((scanIndex == endIndex) || format.Length < 3) 
```

Comment:
`scanIndex == endIndex` Length==0? It's covered by second condition already.

Reply:
removed the condition and checked-in.